custom_components/ha-glocaltokens/api.py

```diff
@@ -46,3 +57,69 @@ def get_google_devices_json(self):
     def get_android_id(self):
         """Generate random android_id"""
         return self._client._get_android_id()
+
+    def format_timer_information(timer):
+        timestamp_ms = timer[FIRE_TIME]
+        timestamp = timestamp_ms / 1000
+        humantime = datetime.fromtimestamp(timestamp).strftime(SHOW_TIME_ONLY)
+        timer[DATE_TIME] = humantime
+
+        duration_ms = timer[ORIGINAL_DURATION]
+        duration = duration_ms / 1000
+        humanduration = datetime.utcfromtimestamp(duration).strftime(SHOW_TIME_ONLY)
+        timer[DURATION] = humanduration
+        return timer
+
+    def format_alarm_information(alarm):
+        timestamp_ms = alarm[FIRE_TIME]
+        timestamp = timestamp_ms / 1000
+        humantime = datetime.utcfromtimestamp(timestamp).strftime(SHOW_DATE_TIMEZONE)
+        localtime = datetime.fromtimestamp(timestamp).strftime(SHOW_DATE_AND_TIME)
+        alarm[DATE_TIME] = humantime
+        alarm[LOCAL_TIME] = localtime
+        return alarm
+
+    def create_url(ip):
+        url = 'https://'+ ip +':'+ str(PORT) + ENDPOINT
+        return url
+
+    def get_alarms_and_timers_from(ip, token):
+
+        url = GlocaltokensApiClient.create_url(ip)
+
+        HEADERS[HEADER_CAST_LOCAL_AUTH] = token
+
+        response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
+
+        return response
+
+    def get_google_devices_information(self):
+
+        devices = self._client.get_google_devices_json()
+
+        for x in range(len(devices)):
+            #local_token = next(item[GLOCALTOKENS_TOKEN] for item in devices)
+            local_token = devices[1][GLOCALTOKENS_TOKEN]
+
+            response = GlocaltokensApiClient.get_alarms_and_timers_from('192.168.0.205', local_token) # IP
+
+            if response.status_code != HTTP_OK:
+                _LOGGER.error("API returned {}".format(response.status_code))
+                #_LOGGER.error(local_token)
+                return devices
+
+            result = response.json()
+
+            if GLOCALTOKENS_TIMERS not in result and GLOCALTOKENS_ALARMS not in result:
+                _LOGGER.error(API_RETURNED_UNKNOWN)
+                return devices
+
+            devices[x][GLOCALTOKENS_TIMERS] = result[GLOCALTOKENS_TIMERS]
+            for y in range(len(result[GLOCALTOKENS_TIMERS])):
+                devices[x][GLOCALTOKENS_TIMERS][y] = GlocaltokensApiClient.format_timer_information(result[GLOCALTOKENS_TIMERS][y])
```

Comment:
What are we doing here? Is it just to iterate through all the timers and replace them with human readable format? 
If so, you could achieve the same in a bit cleaner way like so 
`timers = [self.format_timer_information(timer) for timer in timers]`